src/test/java/part1/exercise/StreamsExercise1.java

```diff
@@ -25,14 +26,50 @@
 
     @Test
     public void getAllEpamEmployees() {
-        List<Person> epamEmployees = null;// TODO all persons with experience in epam
-        throw new UnsupportedOperationException();
+        // TODO all persons with experience in epam
+        List<Employee> employeeList = Generator.generateEmployeeList();
+        List<Person> personsWithExperienceInEpam = employeeList.stream()
+                .filter(e -> e.getJobHistory().stream()
+                        .anyMatch(j -> j.getEmployer().equals("epam")))
+                .map(Employee::getPerson)
+                .collect(toList());
+
+        List<Person> expected = new ArrayList<>();
+        for (Employee employee : employeeList) {
+            inner:
+            for (JobHistoryEntry jobHistoryEntry : employee.getJobHistory()) {
+                if (jobHistoryEntry.getEmployer().equals("epam")) {
+                    expected.add(employee.getPerson());
+                    break inner;
+                }
+            }
+        }
+
+        assertEquals(personsWithExperienceInEpam.size(), expected.size());
+        assertTrue(personsWithExperienceInEpam.containsAll(expected));
     }
 
     @Test
     public void getEmployeesStartedFromEpam() {
-        List<Person> epamEmployees = null;// TODO all persons with first experience in epam
-        throw new UnsupportedOperationException();
+        // TODO all persons with first experience in epam
+        List<Employee> employeeList = Generator.generateEmployeeList();
+        List<Person> personsWithFirstExperienceInEpam = employeeList.stream()
+                .filter(
+                        e -> !e.getJobHistory().isEmpty()

```

Comment:
You could use `e.getJobHistory().stream().take(1).map(...).anyMatch(...)` instead.